### DISCUSSION TITLE
Problems using a trained model with custom tags

### DISCUSSION THREAD

**Discussion by d4rkm0nkey** [GRAAL-Research/deepparse#186] [General]
I trained my model with 11 custom tags like this:
```
address_parser.retrain(
    training_container,
    train_ratio=0.8,
    epochs=50,
    batch_size=128,
    name_of_the_retrain_parser="DetailedParser",
    prediction_tags=address_components,
    callbacks=[lr_scheduler]
)
```
The training works, but if I want to load the last checkpoint like this:
```
address_parser = AddressParser(
    path_to_retrained_model="checkpoints/checkpoint_epoch_47.ckpt"
)
```
I get the error `size mismatch for decoder.linear.bias: copying a param with shape torch.Size([11]) from checkpoint, the shape in current model is torch.Size([9])`.

Can anyone help me, how I can change the total number of tags?

**Comment by davebulaval** [comment]
I see the problem here. When we retrain an AddressParser, at the end, we create a checkpoint with all the metadata, such as the tags, name of the parser and other important information to allow reloading. By default, the checkpoint is named `"retrain_modeltype_address_parser.ckpt"`. In your case, since you have stated a name, the checkpoint will use the name `"DetailedParser"` (i.e. `"retrained_DetailedParser_address_parser.ckpt"`.

I have improved error handling for that specific case in dev. It will be released in 0.9.6.

**Reply by davebulaval** [reply]
@d4rkm0nkey I have pushed an improvement in dev that give more insight in that specific case. It will be released in 0.9.6 soon. I hope it helps.
